server/src/main/java/org/opensearch/transport/TransportService.java

```diff
@@ -391,6 +391,10 @@ public void connectToNode(DiscoveryNode node) throws ConnectTransportException {
         connectToNode(node, (ConnectionProfile) null);
     }
 
+    public void connectToNode(final DiscoveryNode node, boolean skipValidation) {
```

Comment:
Let's add a comment here explaining about the method and why are we skipping the validation for extensibility